src/mapping/AxialGeoMultiscaleMapping.cpp

```diff
@@ -0,0 +1,175 @@
+}
+
+void AxialGeoMultiscaleMapping::mapConsistent(const time::Sample &inData, Eigen::VectorXd &outData)
+{
+  PRECICE_TRACE();
+
+  const int              inValueDimensions = inData.dataDims;
+  const Eigen::VectorXd &inputValues       = inData.values;
+  Eigen::VectorXd &      outputValues      = outData;
+  // TODO: check if this needs to change when access to mesh dimension is possible
+  const int outValueDimensions = outData.size() / output()->vertices().size();
+
+  int effectiveCoordinate = _axis;
+  PRECICE_ASSERT(effectiveCoordinate == MultiscaleAxis::X ||
+                     effectiveCoordinate == MultiscaleAxis::Y ||
+                     effectiveCoordinate == MultiscaleAxis::Z,
+                 "Unknown multiscale axis type.")
+
+  PRECICE_ASSERT((inputValues.size() / inValueDimensions == static_cast<int>(input()->vertices().size())),
+                 inputValues.size(), inValueDimensions, input()->vertices().size());
+  PRECICE_ASSERT((outputValues.size() / outValueDimensions == static_cast<int>(output()->vertices().size())),
+                 outputValues.size(), outValueDimensions, output()->vertices().size());
+
+  PRECICE_DEBUG("Map consistent");
+  if (_type == SPREAD) {
+    /*
+      3D vertices are assigned a value based on distance from the 1D vertex.
+      Currently, a Hagen-Poiseuille profile determines the velocity value.
+    */
+    PRECICE_ASSERT(input()->vertices().size() == 1);
+    mesh::Vertex &v0      = input()->vertices()[0];
+    size_t const  outSize = output()->vertices().size();
+
+    for (size_t i = 0; i < outSize; i++) {
+      PRECICE_ASSERT(static_cast<int>((i * outValueDimensions) + effectiveCoordinate) < outputValues.size(), ((i * outValueDimensions) + effectiveCoordinate), outputValues.size());
+      // When adding support for 2D, remember that this should be 1.5 * inputValues(effectiveCoordinate) * (1 - (_vertexDistances[i] * _vertexDistances[i]));
+      outputValues((i * outValueDimensions) + effectiveCoordinate) = 2 * inputValues(effectiveCoordinate) * (1 - (_vertexDistances[i] * _vertexDistances[i]));
+    }
+  } else {
+    PRECICE_ASSERT(_type == COLLECT);
+    /*
+      1D vertex is assigned the averaged value over all 3D vertices at the outlet,
+      but only of the effectiveCoordinate component of the velocity vector.
+    */
+    PRECICE_ASSERT(output()->vertices().size() == 1);
+    outputValues(effectiveCoordinate) = 0;
+    size_t const inSize               = input()->vertices().size();
+    for (size_t i = 0; i < inSize; i++) {
+      PRECICE_ASSERT(static_cast<int>((i * inValueDimensions) + effectiveCoordinate) < inputValues.size(), ((i * inValueDimensions) + effectiveCoordinate), inputValues.size())
+      outputValues(effectiveCoordinate) += inputValues((i * inValueDimensions) + effectiveCoordinate);
+    }
+    outputValues(effectiveCoordinate) = outputValues(effectiveCoordinate) / inSize;
+  }
+}
+
+void AxialGeoMultiscaleMapping::tagMeshFirstRound()
```

Comment:
Do we safeguard somewhere against using this feature in parallel? Should be easy in the configuration through the `ConfigurationContext.`

Reply:
Probably you didn't adjust the `Participant` construction:

```diff
- Participant cplInterface(context.name, context.config(), 0, 1);
+ Participant cplInterface(context.name, context.config(), context.rank, context.size);
```

Actually, better to use this in any integration test.